actionpack/lib/action_dispatch/system_testing/driver.rb

```diff
@@ -9,23 +9,43 @@ def initialize(name, **options)
       end
 
       def use
-        register if selenium?
+        case @name
+        when :selenium then register_selenium
+        when :poltergeist then register_poltergeist
+        when :webkit then register_webkit
+        end
+
```

Comment:
Can you make a new private register method and move this case statement inside. Then move the duplicate code from the namespaced register methods into the register method?

```ruby
def use
  register unless rack_test?
  setup
end

private
  def register
   Capybara.register_driver @name do |app|
    case @name
    etc
  end
end

def register_selenium
...
end

etc
```